### PR TITLE
Fix/aapd 1854 submission email

### PR DESCRIPTION
**Pull request by topickering**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-1854

## Description of change

Creates a new route and checked log-in redirect for downloading pdf docs in v2 appeal submissions

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: docker-compose.full.yml [hunk 1]
```diff
@@ -16,13 +16,25 @@ services:
       - 3310:3310
     restart: unless-stopped
 
+  # pdf-service-api:
```
Comment (Si3rr4wow) [inline docker-compose.full.yml:19]: you did the thing

Comment (topickering) [reply docker-compose.full.yml:19]: Whoops! Reinstated

File: packages/forms-web-app/src/controllers/document.js [hunk 2]
```diff
@@ -73,6 +80,59 @@ const getDocument = async (req, res) => {
...
 
+/**
+ * links user to a document, requires an active session
+ * currently only used in links to pdf for has and s78 submission
+ * @type {import('express').Handler}
+ */
+const getDocumentV2 = async (req, res) => {
+	const { appealOrQuestionnaireId, documentId } = req.params;
+
+	if (appealOrQuestionnaireId.includes('../') || documentId.includes('../')) {
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/controllers/document.js:91]: Why would these things have that in them?

Comment (topickering) [reply packages/forms-web-app/src/controllers/document.js:91]: They _shouldn't_ have them in, but in theory an appellant could set these params and try and instigate a path traversal attack.

Comment (bryn-pins) [reply packages/forms-web-app/src/controllers/document.js:91]: I'll be honest I'm on the side of checking they are uuids...
also maybe best to do in the handler

File: packages/forms-web-app/src/controllers/common/enter-code.js [hunk 3]
```diff
@@ -215,6 +217,10 @@ const postEnterCode = (views, { isGeneralLogin = true }) => {
 			);
 		}
 
+		if (isV2DocRequest) {
+			return handleCustomRedirect();
```
[the remark this answers is not in the document]
Comment (bryn-pins) [reply packages/forms-web-app/src/controllers/common/enter-code.js:221]: ah it already is sorry

File: packages/forms-web-app/src/middleware/check-logged-in.js [hunk 4]
```diff
@@ -51,6 +57,10 @@ const checkLoggedIn = async (req, res, next) => {
 			return res.redirect(docRedirectUrl);
 		}
 
+		if (v2docRedirect) {
+			req.session.loginRedirect = req.originalUrl;
+		}
+
 		req.session.newOrSavedAppeal = NEW_OR_SAVED_APPEAL_OPTION.RETURN;
```
[the remark this answers is not in the document]
Comment (topickering) [reply packages/forms-web-app/src/middleware/check-logged-in.js:64]: I think we do - this path is used if the user isn't logged in and then gets redirected to the enter email address page, which leads to enter-code after which we use the loginRedirect.

File: packages/forms-web-app/src/controllers/document.js (hunk 2 above)
Comment (topickering) [reply packages/forms-web-app/src/controllers/document.js]: Yeah, seems safer than trying to do string includes or regex, so have added a check in the fetch step

**Review by bryn-pins — APPROVED** [review]

**Review by bryn-pins — APPROVED** [review]

**Pull request merged**